For issue #219, fix ReSpec errors. Plus some other changes.

aShinjiroUrata opened the pull request:
To fix ReSpec errors:
- change order of some of sections and added subsections.
- added tables to explain attributes, parameters, members of interfaces, dictionaries
- added dfn, data-dfn-for, data-link-for
Other changes
- VISClient constructor definition changed(added overload)
- Minor fix (fixed some words, sentences)

aShinjiroUrata committed:
For issue #219, fix ReSpec errors. Plus some other changes.
To fix ReSpec errors:
- change order of some of sections and added subsections.
- added tables to explain attributes, parameters, members of interfaces, dictionaries
- added dfn, data-dfn-for, data-link-for
Other changes
- VISClient constructor definition changed(added overload)
- Minor fix (fixed some words, sentences)

aShinjiroUrata commented:
For resolving ReSpec error and clean up purpose, I created this PR.
(Although this looks not a small change, this is basically a format change and contents change is small)

Could someone review this? @wonsuk73 , @acrofts84 , @rstreif ?

lgewtc commented on vehicle_information_api/vehicle_information_api_specification.html:
```diff
@@ -141,14 +141,14 @@
     <section id="terminology">
       <h2>Terminology</h2>
-      <p> The term 'VISS' is used to refer to the 'Vehicle Information Service Specification' (see <a href="https://w3c.github.io/automotive/vehicle_data/vehicle_information_service.html">here</a>) which is in the process of developed in W3C Automotive Working Group along with this specification.
+      <p> The term <dfn>'VISS'</dfn> is used to refer to the '<dfn>Vehicle Information Service Specification</dfn>' (see <a href="https://w3c.github.io/automotive/vehicle_data/vehicle_information_service.html">here</a>) which is in the process of development in W3C Automotive Working Group along with this specification.
```
<dfn> element is used for term's definition like mostly attribute and methods in WebIDL as well as unfamiliar terminologies. In the case of VISS, VIAS, VSS, I think no need <dfn> element. you can see how <dfn> element can be used as the follow link. https://github.com/w3c/respec/wiki/User's-Guide#definitions-and-linking

In addition, some external specifications could be referred using [[FOO]] notation. VISS and VIAS are already registered in Specref, so if you put [[vehicle-information-service]] within the VIAS spec, the proper link is automatically applied and the reference would be also added in Informative references section at the bottom of the spec document. In Specref, you can find several related specifications like W3C, WHATWG, IETF. (c.f. In case of GENIVI you can add the reference as the same style manually like a square bracket and underline). https://www.specref.org

anawhj commented on vehicle_information_api/vehicle_information_api_specification.html:
```diff
@@ -141,14 +141,14 @@
     <section id="terminology">
       <h2>Terminology</h2>
-      <p> The term 'VISS' is used to refer to the 'Vehicle Information Service Specification' (see <a href="https://w3c.github.io/automotive/vehicle_data/vehicle_information_service.html">here</a>) which is in the process of developed in W3C Automotive Working Group along with this specification.
+      <p> The term <dfn>'VISS'</dfn> is used to refer to the '<dfn>Vehicle Information Service Specification</dfn>' (see <a href="https://w3c.github.io/automotive/vehicle_data/vehicle_information_service.html">here</a>) which is in the process of development in W3C Automotive Working Group along with this specification.
```
&lt;dfn&gt; element is used for term's definition like mostly attribute and methods in WebIDL as well as unfamiliar terminologies. In the case of VISS, VIAS, VSS, I think no need &lt;dfn&gt; element. you can see how &lt;dfn&gt; element can be used as the follow link. https://github.com/w3c/respec/wiki/User's-Guide#definitions-and-linking

In addition, some external specifications could be referred using [[FOO]] notation. VISS and VIAS are already registered in Specref, so if you put [[vehicle-information-service]] within the VIAS spec, the proper link is automatically applied and the reference would be also added in Informative references section at the bottom of the spec document. In Specref, you can find several related specifications like W3C, WHATWG, IETF. (c.f. In case of GENIVI you can add the reference as the same style manually with a square bracket and an underline). https://www.specref.org

aShinjiroUrata replied:
Thanks, understood.
I have fixed this and commit/pushed.

anawhj commented on vehicle_information_api/vehicle_information_api_specification.html:
```diff
@@ -180,11 +180,12 @@
 
       <pre class="idl">
-      [Constructor(optional VISClientOptions options)]
+      [Constructor,
```
Looks good to me. (Sometimes the variable name `options` could get confused with `optional` keyword, but it would be changed in the next work as you mentioned)

aShinjiroUrata replied:
Thanks. Understood.

anawhj commented on vehicle_information_api/vehicle_information_api_specification.html:
```diff
@@ -225,323 +226,445 @@
       };
       </pre>
 
-      <h3>Constructor</h3>
```
As VISClient has multiple constructors, some sentences seems a little strange within this paragraph.

anawhj commented on vehicle_information_api/vehicle_information_api_specification.html:
```diff
@@ -225,323 +226,445 @@
-        </tr>
-      </table>
+      <section id="visclient-constructor">
+        <h3>Constructor</h3>
```
As VISClient has multiple constructors, some sentences seems a little strange within this paragraph.

aShinjiroUrata replied:
I have revised the sentences.

anawhj commented on vehicle_information_api/vehicle_information_api_specification.html:
```diff
@@ -594,27 +717,29 @@
+          <td><code><a>VISSubscribeFilters</a></code></td>
           <td>
-            object containing the filter conditions passed to the <code>subscribe()</code> method. Structure of this object depends on underlying server's specification.<br>
+            Nullable<br>
```
I wonder What does `Nullable` mean.

aShinjiroUrata replied:
To look at other specs, define attributes, members as 'nullable' (by '?' notation) looks common practice.
https://www.w3.org/TR/webrtc/#interface-definition
Could you elaborate your concern?

anawhj commented on vehicle_information_api/vehicle_information_api_specification.html:
```diff
@@ -572,7 +695,7 @@
     </section>
 
-    <section id="subscription-object" data-dfn-for="VISSubscription" data-link-for="VISSubscription">
+    <section id="vissubscription-interface" data-dfn-for="VISSubscription" data-link-for="VISSubscription">
```
As far as I know, a VISSubscription instance is generated only from the return value of subscribe method in VISClient. If it's true, the constructor of VISSubscription could be eliminated.

aShinjiroUrata replied:
I agree. I have deleted 'constructor'.

anawhj commented on vehicle_information_api/vehicle_information_api_specification.html:
```diff
@@ -659,7 +852,7 @@
     </section>
 
-    <section id="vss-object" data-dfn-for="VSS" data-link-for="VSS">
+    <section id="vss-interface" data-dfn-for="VSS" data-link-for="VSS">
```
As same of `VISSubscription` case above, the constructor of VSS could be eliminated.

aShinjiroUrata replied:
I agree. I have deleted 'constructor'.

anawhj commented on vehicle_information_api/vehicle_information_api_specification.html:
```diff
@@ -225,323 +226,445 @@
-      </table>
+      <section id="visclient-constructor">
+        <h3>Constructor</h3>
+        <p>The <code>VISClient</code> constructor, when invoked, must return a new VISClient object.
```
Some `function` keyword could be changed to `method`.

aShinjiroUrata replied:
I'll fix it.

anawhj commented:
The structure of `VIAS` Specification seems better than before. As you requested from #214, I briefly review the changes and put some comments.

aShinjiroUrata commented:
Hyojin san,
Thanks for the great comments!
I'll check and fix them.

aShinjiroUrata committed:
For issue #219: 2nd change. As response to comments in PullRequest #220.
- Revised use of <dfn> and removed unnecessary use.
- Revised description sentenses in '5.1 Constructor'.
- Removed '[Constructor]' from 'VISSubscription' and 'VSS' definition
- Updated some of keywords 'function' to 'method'.

aShinjiroUrata commented:
Hyojin-san, I'm so sorry that the response is so late.
I revised ,mostly according to your comment.

anawhj commented:
It seems great work!
I think this PR could be merged in VIAS, if there is no any other comments.

aShinjiroUrata commented:
@anawhj -san, thanks for review again.
I have merged the change.
